### PR TITLE
ARROW-18278: [Java] Adjust path in Maven generate-libs-jni-macos-linux

### PR DESCRIPTION
**Pull request by rok**
This is a follow-up of ARROW-18042.

### CODE REVIEW

**Comment by rok** [comment]
@github-actions crossbow submit preview-docs

**Comment by rok** [comment]
@github-actions crossbow submit preview-docs

**Comment by rok** [comment]
@github-actions crossbow submit java-jars

**Comment by rok** [comment]
Docs changes are [here](http://crossbow.voltrondata.com/pr_docs/14623/developers/java/building.html#building-jni-libraries-dylib-so-dll).

File: docs/source/developers/java/building.rst [hunk 1]
```diff
@@ -132,7 +132,12 @@ Maven
         $ cd arrow/java
         $ export JAVA_HOME=<absolute path to your java home>
         $ java --version
-        $ mvn generate-resources -Pgenerate-libs-jni-macos-linux -N
+        $ mvn generate-resources \
+            -Pgenerate-libs-jni-macos-linux \
+            -DARROW_GANDIVA=ON \
+            -DARROW_JAVA_JNI_ENABLE_GANDIVA=ON \
+            -DARROW_JAVA_JNI_ENABLE_C=ON \
```
Comment (kou) [inline docs/source/developers/java/building.rst:139]: Should we add this?
This item is described as "To build all JNI libraries (MacOS / Linux) except the JNI C Data Interface library".

Comment (rok) [reply docs/source/developers/java/building.rst:139]: How about if we change the description? Without this change the build always fails for me.

Comment (kou) [reply docs/source/developers/java/building.rst:139]: I'm OK with changing the description. (We need to update "To build only the JNI C Data Interface library (MacOS / Linux)" and related items too with the approach.)

@davisusanibar What do you think about this?

BTW, if you run the "To build only the JNI C Data Interface library (MacOS / Linux)" item separately, can we build without error?

Comment (rok) [reply docs/source/developers/java/building.rst:139]: > I'm OK with changing the description. (We need to update "To build only the JNI C Data Interface library (MacOS / Linux)" and related items too with the approach.)

Ok, I'll push a change.

> BTW, if you run the "To build only the JNI C Data Interface library (MacOS / Linux)" item separately, can we build without error?

```
mvn clean install
mvn generate-resources -Pgenerate-libs-cdata-all-os -N
mvn -Darrow.c.jni.dist.dir=$PWD/../java-dist/lib -Parrow-c-data clean install
```
builds fine on master on my M1.

Comment (kou) [reply docs/source/developers/java/building.rst:139]: Ah, sorry. I meant the followings:

```bash
mvn clean install
rm -rf ../java-dist ../java-jni
mvn generate-resources -Pgenerate-libs-cdata-all-os -N
mvn generate-resources -Pgenerate-libs-jni-macos-linux -DARROW_GANDIVA=ON -DARROW_JAVA_JNI_ENABLE_GANDIVA=ON -N
mvn -Darrow.cpp.build.dir=$PWD/../java-dist/lib/ -Darrow.c.jni.dist.dir=$PWD/../java-dist/lib/ -Parrow-jni clean install
```

Comment (rok) [reply docs/source/developers/java/building.rst:139]: These steps pass for me on this branch and fail on master (on `generate-libs-jni-macos-linux` step).

Comment (kou) [reply docs/source/developers/java/building.rst:139]: Then, how about removing the `-DARROW_JAVA_JNI_ENABLE_C=ON` from this branch?
Building `libarrow_cdata_jni.dylib` and other `.dylib`s for JNI separately is the story used in the current document.

Comment (rok) [reply docs/source/developers/java/building.rst:139]:
```suggestion
```

Comment (rok) [reply docs/source/developers/java/building.rst]: Sure! Removed.

**Review by kou — APPROVED** [review]
+1

**Pull request merged**